clients/dns-service-client/src/lib.rs

```diff
@@ -89,62 +91,3 @@ pub fn is_retryable(error: &DnsConfigError<crate::types::Error>) -> bool {
 
     false
 }
-
-type DnsRecords = HashMap<String, Vec<types::DnsRecord>>;
-
-impl types::DnsConfigParams {
-    /// Given a high-level DNS configuration, return a reference to its sole
-    /// DNS zone.
-    ///
-    /// # Errors
-    ///
-    /// Returns an error if there are 0 or more than one zones in this
-    /// configuration.
-    pub fn sole_zone(&self) -> Result<&types::DnsConfigZone, anyhow::Error> {
-        ensure!(
-            self.zones.len() == 1,
-            "expected exactly one DNS zone, but found {}",
-            self.zones.len()
-        );
-        Ok(&self.zones[0])
-    }
-}
-
-impl Ord for types::DnsRecord {
-    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
-        use types::DnsRecord;
-        match (self, other) {
-            // Same kinds: compare the items in them
-            (DnsRecord::A(addr1), DnsRecord::A(addr2)) => addr1.cmp(addr2),
-            (DnsRecord::Aaaa(addr1), DnsRecord::Aaaa(addr2)) => {
-                addr1.cmp(addr2)
-            }
-            (DnsRecord::Srv(srv1), DnsRecord::Srv(srv2)) => srv1
-                .target
-                .cmp(&srv2.target)
-                .then_with(|| srv1.port.cmp(&srv2.port)),
-
-            // Different kinds: define an arbitrary order among the kinds.
-            // We could use std::mem::discriminant() here but it'd be nice if
-            // this were stable over time.
-            // We define (arbitrarily): A < Aaaa < Srv
-            (DnsRecord::A(_), DnsRecord::Aaaa(_) | DnsRecord::Srv(_)) => {
-                std::cmp::Ordering::Less
-            }
-            (DnsRecord::Aaaa(_), DnsRecord::Srv(_)) => std::cmp::Ordering::Less,
-
-            // Anything else will result in "Greater".  But let's be explicit.
-            (DnsRecord::Aaaa(_), DnsRecord::A(_))
-            | (DnsRecord::Srv(_), DnsRecord::A(_))
-            | (DnsRecord::Srv(_), DnsRecord::Aaaa(_)) => {
-                std::cmp::Ordering::Greater
-            }
-        }
-    }
-}
```

Comment:
This `Ord` impl should be identical to the automatically-derived one, since that orders stuff from top to bottom (so `A`, `AAAA`, and then `SRV`).